packages/smooth_app/lib/cards/data_cards/image_upload_card.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:io';
+import 'package:flutter/material.dart';
+
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:openfoodfacts/model/Product.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:image_cropper/image_cropper.dart';
+
+class ImageUploadCard extends StatefulWidget {
+  const ImageUploadCard({
+    Key key,
+    this.product,
+    this.imageField,
+    this.buttonText,
+  }) : super(key: key);
+
+  final Product product;
+  final ImageField imageField;
+  final String buttonText;
+
+  @override
+  _ImageUploadCardState createState() => _ImageUploadCardState();
+}
+
+class _ImageUploadCardState extends State<ImageUploadCard> {
+  File _image;
+  final ImagePicker picker = ImagePicker();
+
+  Future<void> _getImage() async {
+    print('getImage');
+    final PickedFile pickedFile =
+        await picker.getImage(source: ImageSource.camera);
+    print('got image');
+
+    if (pickedFile != null) {
+      final File croppedImage = await ImageCropper.cropImage(
+        sourcePath: pickedFile.path,
+        androidUiSettings: const AndroidUiSettings(
+          lockAspectRatio: false,
+          hideBottomControls: true,
+        ),
+      );
+      print('cropping image - done');
+
+      if (croppedImage != null) {
+        setState(() {
+          _image = croppedImage;
+        });
+
+        final SendImage image = SendImage(
+          lang: OpenFoodFactsLanguage.ENGLISH,
+          barcode: widget.product.barcode,
+          imageField: widget.imageField,
+          imageUrl: _image.uri,
+        );
+
+        // a registered user login for https://world.openfoodfacts.org/ is required
+        const User myUser =
```

Comment:
We should put the definition of this `User` somewhere, to be reused until we implement the login/password in smoothie. Perhaps in `UserPreferencesModel`?

Reply:
At some point, I'd like to move the UserPreferencesModel to openfoodfacts-dart (as well as the code in match.dart), so that the personalized ranking with attributes can be reused in other apps. So I would prefer not to put the User there. Maybe somewhere in main.dart in the mean time?

Reply:
@stephanegigandet Just put it in `UserPreferencesModel` for the moment. Anyway, we won't be able to move the whole `UserPreferencesModel` as is in openfoodfacts-dart (cf. #140) - your `User` will be part of what remains in smoothie.